libheif/heif_file.cc

```diff
@@ -27,6 +27,11 @@
 #include <cstring>
 #include <cassert>
 
+#ifdef _MSC_VER
+#define NOMINMAX
```

Comment:
heif_file.cc:31: warning: "NOMINMAX" redefined
   31 | #define NOMINMAX
      |
In file included from c:\msys1021\include\c++\10.2.1\x86_64-w64-mingw32\bits\c++config.h:522,
                 from c:\msys1021\include\c++\10.2.1\cstdint:38,
                 from c:\msys1021\include\c++\10.2.1\cinttypes:38,
                 from box.h:28,
                 from heif_file.h:28,
                 from heif_file.cc:21:
c:\msys1021\include\c++\10.2.1\x86_64-w64-mingw32\bits\os_defines.h:45: note: this is the location of the previous definition
   45 | #define NOMINMAX 1

Reply:
Thanks! Easy enough to fix, but I'm concerned that MinGW is defining `_MSC_VER`. It shouldn't be doing that. Do you get an error in `HeifFile::read_from_file` too?

Reply:
Also please give your mingw-64 version.

Reply:
I am using version 10.2.1 for the x265 codec. Unfortunately, there are no upgrades to new versions of GCC and you should create it yourself.
http://msystem.waw.pl/x265/

Reply:
Interestingly, it looks like MinGW does sometimes define `_MSC_VER` as it relates more to the CRT than to Visual Studio itself. Still, that suggests this should all be okay now, as long as you can confirm that `HeifFile::read_from_file` doesn't cause an error for you.

Reply:
For MINGW is
`#if (defined(_MSC_VER) && _MSC_VER >= 1900) || defined(__MINGW32__) || defined(__MINGW64__)`
`#if (defined(_MSC_VER) && _MSC_VER < 1900) || !defined(__MINGW32__) || !defined(__MINGW64__)`
I am amateur, not programmer. This isn't a rule and value 1900 depends on the MSC version and thus how much the library is overdue on github and isn't updated.

Reply:
That's for detecting Visual Studio specifically, not VS CRT compatibility. That's why I asked whether you get an error from `HeifFile::read_from_file`: if you do, then we'll need the full check that you showed; if you don't, then this is already the right code as MinGW is "simulating" Visual Studio for this feature (and its intention to do this in many cases is why it defines `_MSC_VER`).

Reply:
i don't get an error from HeifFile::read_from_file.

Reply:
Excellent. :) Then I think we're good to go!

(not sure of the relevance of avif here?)